cpp/ql/src/jsf/4.13 Functions/AV Rule 114.ql

```diff
@@ -30,7 +30,13 @@ predicate functionsMissingReturnStmt(Function f, ControlFlowNode blame) {
   ) and
   exists(ReturnStmt s |
     f.getAPredecessor() = s and
-    blame = s.getAPredecessor()
+    (
+      blame = s.getAPredecessor() and
+      count(blame.getASuccessor()) = 1
+      or
```

Comment:
Under what circumstances would `blame != s`?

Reply:
Consider something like this:
```
int myFunc(bool cond)
{
	if (cond)
	{
		doSomething();
	} else {
		doSomethingElse();
	}
	// there's an implicit `return;` here, called `s` in the QL.
}
```
In this case `s` has two predecessors, the calls to `doSomething` and `doSomethingElse`.  Selecting either or both of them is confusing, as it suggests there is something wrong with those calls when in fact the issue is a missing return statement several lines below them.

On the flip side for a simple case:
```
int myFunc(bool cond)
{
	doSomething();
}
```
it was found that selecting `doSomething` is clearer than selecting the closing `}` (this is open to discussion; I don't think anything breaks if we just select the closing `}` in all cases, for simplicity and consistency, but the original query was deliberately designed to select `doSomething` in this case)

Reply:
The example in test `g7` is similar but even worse, due to the presence of macros that look like they are code (this is based on something we found in the wild).